### PR TITLE
fix: add an exec ex 2u max application check to the checkout flow

### PR DESCRIPTION
**Pull request by iloveagent57**
https://2u-internal.atlassian.net/browse/ENT-7059

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 

## Testing instructions
Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

### CODE REVIEW

File: ecommerce/extensions/executive_education_2u/views.py [hunk 1]
```diff
@@ -196,6 +196,14 @@ def _get_checkout_failure_reason(self, request, basket, product):
 
             if not offers_with_remaining_user_balance:
                 return ExecutiveEducation2UCheckoutFailureReason.NO_OFFER_WITH_ENOUGH_USER_BALANCE
+
+            offers_with_remaining_enrollment_space = [
+                offer for offer in offers_with_remaining_user_balance
+                if offer.is_available(user=request.user)
```
Comment (iloveagent57) [inline ecommerce/extensions/executive_education_2u/views.py:202]: https://github.com/django-oscar/django-oscar/blob/dc4577e373bcd7fc8bbf879a9c6948c8d8b918a1/src/oscar/apps/offer/abstract_models.py#L274 defines `is_available()`, which includes a check of 
`self.max_user_applications - self.get_num_user_applications(user) > 0`, but also will check if the number of basket or global applications (enrollment cap) has been reached.

Comment (adamstankiewicz) [reply ecommerce/extensions/executive_education_2u/views.py:202]: Oh, that's a great find on `is_available`!

File: ecommerce/extensions/executive_education_2u/constants.py [hunk 2]
```diff
@@ -2,6 +2,7 @@ class ExecutiveEducation2UCheckoutFailureReason:
     NO_OFFER_AVAILABLE = 'no_offer_available'
     NO_OFFER_WITH_ENOUGH_BALANCE = 'no_offer_with_enough_balance'
     NO_OFFER_WITH_ENOUGH_USER_BALANCE = 'no_offer_with_enough_user_balance'
+    NO_OFFER_WITH_ENOUGH_ENROLLMENT_SPACE = 'no_offer_with_enough_enrollment_space'
```
Comment (adamstankiewicz) [inline ecommerce/extensions/executive_education_2u/constants.py]: totally a nit: `NO_OFFER_WITH_REMAINING_ENROLLMENTS`? The word "space" is throwing me off for some reason haha.

**Review by adamstankiewicz — APPROVED** [review]

**Comment by adamstankiewicz** [comment]
@iloveagent57 Ah, looks like there's a CI test failure.

**Pull request merged**